### PR TITLE
SEAT-180 Add jest configuration and initial test

### PR DESCRIPTION
**Pull request by josephspens**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,38 +21,55 @@
     "analyze": "webpack --profile --json --progress > stats.json",
     "check-types": "tsc --noEmit",
     "lint": "eslint . --ext js --ext jsx --ext ts --ext tsx --cache",
-    "fix": "npm run lint -- --fix"
+    "fix": "npm run lint -- --fix",
+    "test": "jest",
+    "test:watch": "jest --watch"
+  },
+  "jest": {
```
Comment (mayavera) [inline package.json]: We're already using separate files for our other configs (`.babelrc`, `.eslintrc`, `.stylelintrc`, `tsconfig.json`). Would you mind creating a `jest.config.js` file?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
+// For a detailed explanation regarding each configuration property, visit:
```
Comment (mayavera) [inline jest.config.js]: Mind if we nix these file comments? Jest configuration is relatively straightforward and self-descriptive. Their documentation website should be able to clarify any confusion.

Comment (josephspens) [reply jest.config.js]: This was the file generated from `jest --init`, is there a reason to get rid of it?

Comment (mayavera) [reply jest.config.js]: Honestly, reading a wall of comments to find the right place to change a config stresses me out. The jest documentation does a much better job at conveying the meaning and implications of each setting than a single ascii line.

**Pull request merged**